src/Codeception/Module/MongoDb.php

```diff
@@ -158,7 +158,7 @@ private function validateDump()
                         "Tar gunzip archives are not supported for Windows systems"
                     );
                 }
-                if (strlen($this->dumpFile) <= 7 || (substr($this->dumpFile, -7) !== '.tar.gz' && substr($this->dumpFile, -4) !== '.tgz')) {
+                if (!preg_match('/^.+(\.tar\.gz|\.tgz)$/', $this->dumpFile)) {
```

Comment:
`^.` what about space characters?
I think regex can be simplified to just `/(\.tar\.gz|\.tgz)$/`

Reply:
You bet.

Reply:
Yes, sorry, space character is included

Reply:
What you have should be fine so long you do not care about files that are named just ".tgz".  To be honest the check isn't really needed.  The system will puke if you try to untar something thats not a valid archive... I think... I didn't test it explicitly.

Reply:
Yeah, got it, thanks. Once build finishes I'm merging